src/shaders/composite.h

```diff
@@ -75,20 +75,36 @@ vec4 sampleLayer(sampler2D layerSampler, uint layerIdx, vec2 uv, bool unnormaliz
     } else if (get_layer_colorspace(layerIdx) == colorspace_sRGB) {
         color.rgb = srgbToLinear(color.rgb);
 
+        if(u_itmEnable > 0 && c_st2084Output) {
+            color.rgb = convert_primaries(color.rgb, rec709_to_xyz, xyz_to_rec2020);
+            color.rgb = bt2446a_inverse_tonemapping(color.rgb, u_itmSdrNits, u_itmTargetNits);
+        }
         if (checkDebugFlag(compositedebug_Heatmap)) {
-            color.rgb = hdr_heatmap(color.rgb, false, false, c_st2084Output);
+            if(u_itmEnable > 0 && c_st2084Output) {
+                color.rgb = hdr_heatmap(color.rgb, true, true, c_st2084Output);
+            } else {
+                color.rgb = hdr_heatmap(color.rgb, false, false, c_st2084Output);
+            }
         } else {
-            if (c_st2084Output) {
+            if (u_itmEnable == 0 && c_st2084Output) {
                 color.rgb = linearToNits(color.rgb);
                 if (!c_forceWideGammut)
                     color.rgb = convert_primaries(color.rgb, rec709_to_xyz, xyz_to_rec2020);
             }
         }
     } else if (get_layer_colorspace(layerIdx) == colorspace_linear) {
+        if(u_itmEnable > 0 && c_st2084Output) {
```

Comment:
u_itmEnable can be a spec constant `c_itmEnable` that is only set with c_st2084Output.

That would allow you to simplify a bunch of this too, as you could just pass that value to params of hdr_heatmap.

Reply:
changed to be a spec constant with this:
https://github.com/Plagman/gamescope/pull/751/commits/18d75b51175ca6c65028d3bf32658c7b0c0a2efa
further simplification here:
https://github.com/Plagman/gamescope/pull/751/commits/353f13f7cdaba2b5a631c1ca6887c0f446eec6ab